### ISSUE TITLE
Confusing javax.ws.rs.WebApplicationException instead of GPClientException

### ISSUE THREAD

**Issue by novoj** [gopaycommunity/gopay-java-api#13]
In recent versions (we've upgraded to 3.4.8) we started to receive exceptions like this:

```
Could not parse json {"date_issued":"2020-12-01T12:10:54.357+0100","errors":[{"scope":"G","error_code":200,"error_name":"AUTH","message":"Unauthorized access."}]}
javax.ws.rs.WebApplicationException: Could not parse json {"date_issued":"2020-12-01T12:10:54.357+0100","errors":[{"scope":"G","error_code":200,"error_name":"AUTH","message":"Unauthorized access."}]}
        at cz.gopay.api.v3.impl.apacheclient.AbstractImpl.unMarshall(AbstractImpl.java:61)
        at cz.gopay.api.v3.impl.apacheclient.HttpClientPaymentClientImpl.getPaymentInstruments(HttpClientPaymentClientImpl.java:238)
        at cz.gopay.api.v3.AbstractGPConnector.generatePaymentInstruments(AbstractGPConnector.java:288)
```

Instead our code that was developed for earlier versions of this library expected GPClientException. Also to exception message is misleading - JSON is (probably) correct but the gate returned error code.

This issue looks similar to another issue report https://github.com/gopaycommunity/gopay-java-api/issues/12 - but exception differs.

**Comment by FSichinger** [comment]
Fixed in 3.4.9

**Comment by novoj** [comment]
Thanks!
